app/controllers/api/v1/capes_controller.rb

```diff
@@ -1,26 +1,34 @@
 class Api::V1::CapesController < ApiController
+  before_action :authenticate_user!, except: [:index, :show]
 
   def index
     render json: Cape.all
   end
 
   def show
-    render json: Cape.find(params[:id])
+    cape = Cape.find(params[:id])
+    render json: cape
   end
 
- def create
-  cape = Cape.new(cape_params)
-
-  if cape.save
-     render json: cape
-  else
-     render json: { errors: cape.errors.full_messages }
+  def create
+    cape = Cape.new(cape_params)
+    cape.user = current_user
+    if cape.save
+      render json: cape
+    else
+      render json: { errors: cape.errors.full_messages }
+    end
   end
- end
 
- private
+  protected
+  def cape_params
+    params.require(:cape).permit(:name,:fullName,:gender,:affiliation,:intelligence,:strength,:speed,:url, :user)
+  end
 
- def cape_params
-   params.permit(:name,:fullName,:gender,:affiliation,:intelligence,:strength,:speed,:url)
- end
+  def authorize_user
+    if !user_signed_in
+      flash[:notice] = "Please sign in to access this page"
+      redirect_to root_path
+    end
```

Comment:
redirects dont work in an API controller (because its a json request), only normal controllers. flash also only works with standard http requests as well. Something like 
```
if !user_signed_in
      render json: { message: "User must be signed in }
    end
```